[nananakx-x] iP

nananakx-x opened the pull request:
### Duke 🤖

> "Chatbots will be your _new **best** friend_." - Christine Crandell ([Forbes](https://www.forbes.com/sites/christinecrandell/2016/10/23/chatbots-will-be-your-new-best-friend/?sh=1b2fb202a245))

Duke is a Personal Assistant Chatbot that helps user to keep track of various things, including:
1. ✅ To-do tasks
2. 🕓 Deadlines
3. 🎉 Events

The name Duke was chosen as a placeholder name, in honor of Duke, the Java Mascot. The current name of the Chatbot is John,  It's 
- Text-based
- Easy to learn 
- ~FAST~ 🚨 **_SUPER_** FAST to use 

Features:
- [x] 🗓️ Managing tasks 
- [x] ⏳ Managing deadlines 
- [x] 🎪 Managing events 
- [x] 📄 Displaying your list of tasks 
- [x] 🔍 Searching for a specific keyword 

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```  ruby
public static void main(String[] args) {
    new Duke().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

nananakx-x committed:
Level 0. Rename, Greet, Exit

nananakx-x committed:
Level 1. Echo

nananakx-x committed:
Level 2. Add, List

nananakx-x committed:
Level 3. Mark as Done

nananakx-x committed:
Level 4. ToDos, Events, Deadlines

nananakx-x committed:
Level 5 minimal

nananakx-x committed:
Level 5. Handle Errors

nananakx-x committed:
A-TextUiTesting v2

nananakx-x committed:
Level 6. Delete

nananakx-x committed:
Add InvalidNumberException

nananakx-x committed:
Split code in main to different methods

nananakx-x committed:
Level 7. Save

nananakx-x committed:
Level 7. Save

nananakx-x committed:
Level 8. Dates and Times

nananakx-x committed:
Level 8. Dates and Times

nananakx-x committed:
A-MoreOOP: Use More OOP

nananakx-x committed:
A-MoreOOP: Use More OOP

nananakx-x committed:
A-Packages: Organize into Packages

nananakx-x committed:
A-Packages: Organize into Packages

nananakx-x committed:
Merge add-gradle-support

nananakx-x committed:
A-Gradle: Use Gradle

nananakx-x committed:
A-JUnit: Add JUnit Tests

nananakx-x committed:
A-Jar: Create a JAR File

nananakx-x committed:
A-JavaDoc: JavaDoc

nananakx-x committed:
A-JavaDoc: JavaDoc

nananakx-x committed:
A-CodingStandard: Follow the Coding Standard

nananakx-x committed:
Level-9: Find

nananakx-x committed:
A-CodingStandard: Follow the Coding Standard

nananakx-x committed:
Level-9v2: Find

nananakx-x committed:
Level 9v3. Find

nananakx-x committed:
Level 9v4. Find

nananakx-x committed:
Level 9. Find

nananakx-x committed:
Add more OOP

nananakx-x committed:
Add more methods to Ui

nananakx-x committed:
A-CheckStyle: Use CheckStyle

nananakx-x committed:
A-CheckStyle: Use CheckStyle

Tim-Siu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,68 @@
+            ui.promptReply();
+            String command = sc.nextLine();
+
+            if (command.equals("bye")) {
```
Interesting method for checking whether duke should exit or not. LGTM.

Tim-Siu commented on src/main/java/duke/exception/InvalidCommandException.java:
You set up different exceptions. This can be very helpful in the long run. Good job!

Tim-Siu commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+     * @param dateTimeString String by to be converted.
+     * @return LocalDateTime by.
+     */
+    public LocalDateTime convertDateTime(String dateTimeString) {
```
You may consider putting the parsing logic to the parser. I can see your point too.

Tim-Siu commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,250 @@
+     * @param command The user command.
+     * @param taskList The user's list of tasks.
+     */
+    public void processCommand(String command, TaskList taskList) throws Exception {
```
You may consider using switch statement.

Tim-Siu commented on src/test/java/duke/ParserTest.java:
Testing seems to be thorough. Good job!

Tim-Siu reviewed (commented):
LGTM! Code quality seem to be of the highest standards.

Jweng88 commented on src/test/java/duke/StorageTest.java:
I like how you have thoroughly tested the different methods in this file, with the use suitable naming. Good job! 👍

Jweng88 commented on src/main/java/duke/Parser.java:
The naming of the methods are all of the course standard. Looks good to me! 👍

Jweng88 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,68 @@
+ * the Java Mascot. The current name of the Chatbot is John.
+ */
+public class Duke {
+    private Storage storage;
```
Perhaps single line Javadoc comments can be added for the class members?

Jweng88 commented on src/main/java/duke/Ui.java:
I like the clarity of the comments for the class and methods. Good job!

Jweng88 commented on src/main/java/duke/exception/InvalidDeadlineException.java:
I like how you separated the different types of exceptions with suitable naming for easier understanding. Looks good to me! 👍

Jweng88 reviewed (commented):
Looks good to me! Good job with the code standard!

nananakx-x committed:
Level 10. GUI

nananakx-x committed:
Level 10. GUI

nananakx-x committed:
Add prompText and solve ellipsis issue for UI

timleow commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,267 @@
+            return ui.startMessage();
+        }
+
+        if (command.equals("list")) {
```
Perhaps you could use a `switch` statement here instead of multiple `if-else-` clauses?

timleow commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,113 @@
+
+        for (int i = 1; i <= matchingTaskList.getSize(); i++) {
+            // Adding toString() to use the overridden one in duke.task.Task, etc.
+            matchingList.append(i).append(". ").append(matchingTaskList.getTask(i - 1).toString())
```
Could this statement be broken up into a few lines to make the code more readable?

timleow commented on src/main/java/duke/task/Deadline.java:
I like that you have different exception classes for each kind of error encountered! 💯

timleow commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+ * Represents a deadline task that has a description and a due date/time.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```
Could we use a noun for `by` which represents the date of the `deadline`?

timleow commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,51 @@
+ * Represents an event task that has a description, a start date/time, and an end date/time.
+ */
+public class Event extends Task {
+    protected String from;
```
Could we use a noun to identify the starting/ending date instead of `from` and `by`?

timleow reviewed (commented):
The quality of your code was mostly of a high standard! Just note a few things on naming, and also making code with multiple attribute/method calls in a single line more readable for the user. Other than that, LGTM! 💯

nananakx-x committed:
Update the value of the xmlns attribute

ScarletBlanks commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,65 @@
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
```
I like how you separate the imports!! Very cool!

ScarletBlanks commented on src/main/java/duke/Parser.java:
```diff
@@ -22,16 +22,40 @@ public Parser() {
-        for (int i = 1; i <= taskList.getSize(); i++) {
-            // Adding toString() to use the overridden one in duke.task.Task, etc.
-            System.out.println(i + ". " + taskList.getTask(i - 1).toString());
+    public void processCommand(String command, TaskList taskList) throws Exception {
```
agree with the rest, a switch statement would be nice here

ScarletBlanks commented on src/main/java/duke/Ui.java:
```diff
@@ -70,4 +71,33 @@ public void showTaskUnmarked(Task task) {
+    /**
+     * Displays the user's task list.
+     *
+     * @param taskList The task list to be displayed.
```
Clean and concise javadocs

ScarletBlanks reviewed (commented):
Overall, I found your code easy to read for the most part except a few places
where you might need to use switch instead of multiple if statements. Javadocs is clean and concise.

Chen1x commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,31 @@
+package duke;
+
+import java.io.IOException;
```
Very tidy imports!

Chen1x commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,267 @@
+    public String processCommand(String command, TaskList taskList, Storage storage) throws Exception {
+        storage.createFile();
+
+        if (command.equals("start")) {
```
Switch Case 🅱️erhaps?

Chen1x reviewed (commented):
LGTM! Good coding standard!

nananakx-x committed:
Vary color of dialog boxes for user and chatbot

nananakx-x committed:
Add Java assert feature.

This helps to document important assumptions
that should hold at various points in the code.

Assertions are mainly added in Parser to verify
that taskList is not null.

Some parts in Ui are also split into multiple lines
for better readability.

nananakx-x committed:
Improve Code Quality.

The processCommand method in Parser is modified to use
switch, a helper function is also created to avoid
long methods. In addition, whitespace is used to
structure the code logically.

Refactoring is done to improve the code quality where
necessary. This leads to better readability.

nananakx-x committed:
Merge pull request #1 from nananakx-x/branch-A-Assertions

Add Java assert feature.

nananakx-x committed:
Merge master branch to branch-A-CodeQuality

nananakx-x committed:
Merge pull request #2 from nananakx-x/branch-A-CodeQuality

Improve Code Quality.

nananakx-x committed:
Add an extension

nananakx-x committed:
Add status and help commands

nananakx-x committed:
Add minor fix for check style

nananakx-x committed:
Add Increment: A-CI

nananakx-x committed:
Fix GUI image size

Size of user and chatbot images is changed to
60 so that they do not take up too much space
on the screen.

nananakx-x committed:
Fix image size

nananakx-x committed:
Update readme

nananakx-x committed:
Fix readme

nananakx-x committed:
Fix readme outputs
